specification.json

```diff
@@ -848,6 +848,27 @@
             "content": "`PROVIDER_READY` handlers attached after the provider is already in a ready state MUST run immediately.",
             "RFC 2119 keyword": "MUST",
             "children": []
+        },
+        {
+            "id": "Requirement 5.3.4",
+            "machine_id": "requirement_5_3_4",
+            "content": "If the provider's `on context changed` executes with mismatched contexts, `PROVIDER_STALE` handlers MUST run.",
+            "RFC 2119 keyword": "MUST",
+            "children": []
```

Comment:
>#### Requirement 5.3.4
>If the provider's `on context changed` executes with mismatched contexts, `PROVIDER_STALE` handlers **MUST** run.

What does "mismatched context" mean in this case? Every context that is different from the current one? We would emit this event then, and because of 5.3.5 we would get a `PROVIDER_READY` directly after it right like the diagram shows?

```mermaid
sequenceDiagram
    participant C as Client
    participant P as Provider
    participant E as EventEmitter

    C-)P: Set context
    note right of C: This context is mismatching
    P-)E: Dispatch PROVIDER_STALE
    P->>P: run onContextChanged
    P-)E: Dispatch PROVIDER_READY
```

Reply:
I have the same question as @lukas-reining I don't really understand what "mismatched context" means.

Reply:
https://github.com/open-feature/spec/pull/200#discussion_r1280801308

Reply:
@lukas-reining yes, this is the agreement according to the latest commit - https://github.com/open-feature/spec/pull/200/commits/8d773c0ef61d4bfe82db1f76d13537f3f8b4ade4 . We no longer mention `mismatched contexts` and events should fire for any context change.

Reply:
Great🙂